packages/ddex/ingester/parser/ern38x.go

```diff
@@ -615,6 +647,258 @@ func processReleaseNode(rNode *xmlquery.Node, soundRecordings *[]SoundRecording,
+			// Parse territory codes
+			territoryCodes := dealTerms.SelectElements("TerritoryCode")
+			if len(territoryCodes) != 0 {
+				if !containsWorldwideTerritoryCode(dealTerms) {
+					err = fmt.Errorf("no Worldwide <TerritoryCode> found for <DealReleaseReference>s%v", releaseRefs)
+					break
+				}
+			}
+		}
+
+		// Parse validity start date
+		validityStartStr := safeInnerText(dealTerms.SelectElement("ValidityPeriod/StartDate"))
+		if validityStartStr == "" {
+			err = fmt.Errorf("missing required ValidityPeriod/StartDatea for <DealReleaseReference>s%v", releaseRefs)
+			break
+		}
+		validityStart, validityStartErr := time.Parse("2006-01-02", validityStartStr)
+		if validityStartErr != nil {
+			err = fmt.Errorf("error parsing ValidityPeriod/StartDate for <DealReleaseReference>s%v: %s", releaseRefs, validityStartErr)
+			break
+		}
+
+		// Parse price
+		var wholesalePricePerUnit int
+		var priceCurrencyCode string
+		wholesalePricePerUnitNode := dealTerms.SelectElement("PriceInformation/WholesalePricePerUnit")
+		wholesalePricePerUnitStr := safeInnerText(wholesalePricePerUnitNode)
+		if wholesalePricePerUnitNode != nil {
+			priceCurrencyCode = wholesalePricePerUnitNode.SelectAttr("CurrencyCode")
+		}
+		if wholesalePricePerUnitStr != "" && priceCurrencyCode != "USD" {
+			err = fmt.Errorf("unsupported currency code %s for <WholesalePricePerUnit> for <DealReleaseReference>s%v", priceCurrencyCode, releaseRefs)
+		}
+		if wholesalePricePerUnitStr != "" {
+			var wholesalePricePerUnitErr error
+			wholesalePricePerUnit, wholesalePricePerUnitErr = strconv.Atoi(wholesalePricePerUnitStr)
+			if wholesalePricePerUnitErr != nil {
+				err = fmt.Errorf("Error parsing <WholesalePricePerUnit>%s</WholesalePricePerUnit> for <DealReleaseReference>s%v", wholesalePricePerUnitStr, releaseRefs)
+				break
+			}
+		}
+
+		// Add deal info to each release referenced
+		for _, releaseRef := range releaseRefs {
+			// Find corresponding ParsedReleaseElem
+			elem, found := findParsedReleaseElem(release, releaseRef)
+			if !found {
+				err = fmt.Errorf("no release found corresponding to <DealReleaseReference>%s</DealReleaseReference>", releaseRef)
+			}
+
+			for _, useType := range useTypes {
+				if elem.ReleaseType == common.TrackReleaseType {
+					switch useType {
+					case "Stream", "OnDemandStream":
+						err = addStreamingConditionsToReleaseElem(dealTerms, commercialModelType, useType, wholesalePricePerUnit, releaseRef, elem)
+						if err != nil {
+							break
+						}
+					case "PermanentDownload":
+						err = addDownloadConditionsToReleaseElem(dealTerms, commercialModelType, useType, wholesalePricePerUnit, releaseRef, elem)
+						if err != nil {
+							break
+						}
+					default:
+						err = fmt.Errorf("unsupported <UseType>%s</UseType> for <DealReleaseReference>%s</DealReleaseReference>", useType, releaseRef)
+						break
+					}
+				}
+
+				if validityStartStr != "" {
+					elem.ReleaseDate = validityStart
```

Comment:
i wonder if we should call this "ValidityStartDate" in the schema and use that for determining when to publish. and then only use the existing ReleaseDate for informational purposes about when the track was originally supposed to be released. wdyt?

Reply:
good call.

according to the [docs](https://kb.ddex.net/implementing-each-standard/best-practices-for-all-ddex-standards/deals-and-commercial-aspects/communicating-deal-dates/) the deal validity period's start date is just the "street date" on which the resource is to become available on the platform.

would we want to display `GlobalOriginalReleaseDate` or `ReleaseDate` in `releaseDetails` in the UI and fall back to the deal's start date if both are not present?

and if so, are you thinking to add a field to discovery / the sdk? currently it only understands 1 release date as the date to put the resource on the platform and the corr. date to display

Reply:
i'm thinking we can keep the sdk as is, ie use the existing release date from the GlobalOriginalReleaseDate / ReleaseDate field for the UI.

we can add a validity_start field to the ddex schema for pending_releases and the publisher can poll that instead of polling the UI release date like it does currently. that way we'll still avoid publishing until the validity_start is reached.

we can confirm tomorrow about what to do if the validity_start is before the UI release date (it could upload as a "scheduled release" but probably we'll want to just not upload until the current date is >= both validity_start and the UI release date)